### PR TITLE
Set NetRoslyn value for VMR build

### PR DESCRIPTION
**Pull request by mthalman**
Related to https://github.com/dotnet/installer/pull/18238 and https://github.com/dotnet/roslyn/pull/71507.

There are several projects whose `TargetFramework` is set to `NetRoslyn` which is [statically set to `net8.0`](https://github.com/dotnet/roslyn/blob/db94f712c1f6e4d64d2c61921f9e60629ecb41a1/eng/targets/TargetFrameworks.props#L15). This [causes build failures](https://github.com/dotnet/installer/pull/18238#issuecomment-1898830284) in the context of the VMR because it needs to be targeting `net9.0`.

These changes update the value of `NetRoslyn` in the context of building in the VMR so that it will make use of `NetCurrent` (which will be `net9.0` there).

### CODE REVIEW

File: eng/targets/TargetFrameworks.props [hunk 1]
```diff
@@ -49,6 +49,7 @@
         <NetRoslynToolset>$(NetCurrent)</NetRoslynToolset>
         <NetRoslynSourceBuild>$(NetCurrent);$(NetPrevious)</NetRoslynSourceBuild>
         <NetRoslynAll>$(NetCurrent);$(NetPrevious)</NetRoslynAll>
+        <NetRoslyn>$(NetCurrent)</NetRoslyn>
```
Comment (mmitche) [inline eng/targets/TargetFrameworks.props:52]: Does it need to be in the above property group too?

**Review by jaredpar — CHANGES_REQUESTED** [review]
This section of the `TargetFramework.props` is for source build and `$(NetRoslyn)` should not be used in source build. If we find a case where this is blocking source build then our build authoring is incorrect.

**Comment by mthalman** [comment]
> This section of the `TargetFramework.props` is for source build and `$(NetRoslyn)` should not be used in source build. If we find a case where this is blocking source build then our build authoring is incorrect.

Ok, I'll track down which project(s) caused a failure without this patch applied and reply back.

**Comment by mthalman** [comment]
Here's the build error when this patch isn't applied:

```
/vmr/.dotnet/sdk/9.0.100-preview.2.24103.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1005: Assets file '/vmr/src/roslyn/artifacts/obj/Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost/project.assets.json' doesn't have a target for 'net8.0'. Ensure that restore has run and that you have included 'net8.0' in the TargetFrameworks for your project. [/vmr/src/roslyn/src/Workspaces/Core/MSBuild.BuildHost/Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.csproj::TargetFramework=net8.0]
```

From https://github.com/dotnet/installer/pull/18513

**Comment by jaredpar** [comment]
> This is still an issue as explained in https://github.com/dotnet/installer/pull/18752#issuecomment-1961585290.

As [I explained][explain] the approach in this PR is not the right way to solve the problem. 

[explain]: https://github.com/dotnet/roslyn/pull/71704#pullrequestreview-1863834907

**Comment by jaredpar** [comment]
https://github.com/dotnet/roslyn/pull/72248/files

This is what I envisioned the correct approach to be but it's hitting a source build error 🤦

**Comment by mthalman** [comment]
Closing in favor of the changes in https://github.com/dotnet/roslyn/pull/72248.

**Pull request closed without merging**